runhouse/rns/hardware/cluster.py

```diff
@@ -0,0 +1,405 @@
+from typing import List, Optional, Union, Dict
+import subprocess
+import logging
+from pathlib import Path
+import pkgutil
+import contextlib
+
+import sky
+from sky.utils import command_runner
+from sshtunnel import SSHTunnelForwarder, HandlerSSHTunnelForwarderError
+import ray.cloudpickle as pickle
+
+from runhouse.rns.resource import Resource
+from runhouse.rns.packages.package import Package
+from runhouse.grpc_handler.unary_client import UnaryClient
+from runhouse.grpc_handler.unary_server import UnaryService
+from runhouse.rh_config import rns_client, open_grpc_tunnels
+
+logger = logging.getLogger(__name__)
+
+
+class Cluster(Resource):
+    RESOURCE_TYPE = "cluster"
+
+    def __init__(self,
+                 name,
+                 ips: List[str] = None,
+                 ssh_creds: Dict = None,
+                 dryrun=True,
+                 **kwargs  # We have this here to ignore extra arguments when calling from from_config
+                 ):
+        """
+        Args:
+            name: Name of the cluster
+            dryrun:
+        """
+
+        super().__init__(name=name, dryrun=dryrun)
+
+        self.address = ips[0] if ips else None
```

Comment:
I'm a little bit confused about `ips` being a list of ip addresses to pass in, but only the first one is used through `self.address`.

Reply:
also in what situation would we expect `Cluster` to be initialized with `ips=None`?